api_app/analyzers_manager/file_analyzers/cape_sandbox.py

```diff
@@ -0,0 +1,162 @@
+            result = self.__poll_for_result(task_id=task_id)
+            to_respond["response"] = result
+            return to_respond
+
+        elif (
+            list(response_json.get("errors")[0].values())[0]
+            == "Not unique, as unique option set on submit or in conf/web.conf"
+        ):
+
+            status_id = self.__search_by_md5()
+            gui_report_url = self.__base_url + "/analysis/" + status_id
+            report_url = (
+                self.__base_url + "/apiv2/tasks/get/report/" + status_id + "/json"
+            )
+            to_respond["result_url"] = gui_report_url
+
+            try:
+                final_request = self.__session.get(
+                    report_url,
+                )
+            except requests.RequestException as e:
+                raise AnalyzerRunException(e)
+
+            final_json = final_request.json()
+            to_respond["response"] = final_json
+
+            return to_respond
+
+        return response_json
+
+    def __search_by_md5(
+        self,
+    ):
+        db_search_url = self.__base_url + "/apiv2/tasks/search/md5/" + self.md5
+
+        try:
+            q = self.__session.get(db_search_url)
+            q.raise_for_status()
+
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        status_id = str(q.json().get("data")[0].get("id"))
+
+        return status_id
+
+    def __poll_for_result(
+        self,
+        task_id,
+    ):
+        status_api = self.__base_url + "/apiv2/tasks/status/" + str(task_id)
+        for i in range(self.max_tries):
+            try:
+                r = self.__session.get(status_api)
+                if r.status_code == 429:
+                    time.sleep(self.poll_distance)
+                else:
+                    r.raise_for_status()
+            except requests.RequestException as e:
+                raise AnalyzerRunException(e)
+
+            responded_json = r.json()
+            error = responded_json.get("error")
+            data = responded_json.get("data")
+            if not error and (data in ("reported", "completed")):
+                report_url = (
+                    self.__base_url
+                    + "/apiv2/tasks/get/report/"
+                    + str(task_id)
+                    + "/json"
+                )
+                try:
+                    final_request = self.__session.get(
+                        report_url,
+                    )
+                    final_request.raise_for_status()
+                except requests.RequestException as e:
+                    raise AnalyzerRunException(e)
+
+                final_json = final_request.json()
+
+                if final_json.get("error"):
+                    response_ = final_json
+                    return response_
+
+            elif error:
+                raise AnalyzerRunException(error)
+
+            elif data == "running":
+                time.sleep(30)
```

Comment:
this number should not be hardcoded but a configurable variable. See other similar analyzers (when we used `poll_distance`)